### ISSUE TITLE
Incorrect override amounts when using binary expressions

### ISSUE THREAD

**Issue by mattrossman** [pixiv/three-vrm#1484]
When an expression is set to binary with `isBinary: true`, and these expressions also have blended overrides (e.g. `overrideLookAt: "blend"`), their weight is blended continuously instead of discretely when the corresponding expression group (e.g. gaze expression) has nonzero weight. I expect binary expressions to maintain the sharp 0.5 threshold whether using "blend" overrides or not.

I believe this stems from the `overrideBlinkAmount` / `overrideLookAtAmount` / `overrideMouthAmount` calculations not taking into account the `isBinary` flag, e.g.:

https://github.com/pixiv/three-vrm/blob/7a35f7ee456136602d7bc12450b6fe9c247d7f5d/packages/three-vrm-core/src/expressions/VRMExpression.ts#L61-L69

Changing the expression in the 'blend' case to `this.isBinary ? (this.weight <= 0.5 ? 0.0 : 1.0) : this.weight` as is [done elsewhere in the applyWeight logic](https://github.com/pixiv/three-vrm/blob/7a35f7ee456136602d7bc12450b6fe9c247d7f5d/packages/three-vrm-core/src/expressions/VRMExpression.ts#L115) produces the expected discrete blending behavior.

I wonder if this also affects the 'block' override mode, as the binary expression will begin blocking as soon as it has nonzero weight, even though it doesn't visually apply that weight until after the 0.5 weight threshold. So perhaps the 'block' case should look more like:
```ts
const actualWeight = this.isBinary ? (this.weight <= 0.5 ? 0.0 : 1.0) : this.weight;
return 0.0 < actualWeight ? 1.0 : 0.0;
```

**Comment by 0b5vr** [comment]
Woow, that is a factor I haven't considered in the spec layer; thank you.
Let me consider the interaction between them, I'm not sure your PR is working perfect or not.

**Comment by mattrossman** [comment]
A comparison of binary expressions with texture transform binds and "blend" lookAt overrides:


Before | After
---|---
<video src="https://github.com/user-attachments/assets/5ee366e2-91ad-4390-9b37-e09fe7aa5707" /> | <video src="https://github.com/user-attachments/assets/f404edec-bb4b-4a42-9039-6209a839507e" />

Notice before how when lookLeft is applied, texture transforms on the blink expression apply smoothly instead of a sharp binary change, which breaks the use case of expression textures.

**Comment by 0b5vr** [comment]
Thank you for the detailed explanation! We are currently discussing the behavior with the VRM spec team. I'm 90% confident that we can fix it in the desired way, but we must ensure that the behavior can be described as a spec without confusion before we apply this in our implementation.

**Comment by 0b5vr** [comment]
Since vrm-c/vrm-specification#487 has been merged, we will introduce the change #1489 to three-vrm.

**Comment by 0b5vr** [comment]
@mattrossman The change has been released in v3.1.4 :pray:
https://github.com/pixiv/three-vrm/releases/tag/v3.1.4
